### DISCUSSION TITLE
Please add NiceGUI's API documentation to your nicegui.io website

### DISCUSSION THREAD

**Discussion by zouzhe1** [zauberzeug/nicegui#3361] [Ideas / Feature Requests]
Please add NiceGUI's API documentation to your nicegui.io website

**Comment by rodja** [comment]
What do you mean by API documentation?

**Reply by zouzhe1** [reply]
The present website documentation resembles a rudimentary tutorial and is rather scattered, failing to provide a systematized showcase of all the components in NiceGUI. This can be very frustrating for new learners who find it difficult to locate what they're looking for or achieve a comprehensive understanding of all that's on offer. As such, it could be more beneficial to develop something similar to a code API documentation utilizing tools such as Sphinx or MkDocs.

For instance, when I want to use menus to implement multi-page switching, and also freely adjust text sizes, positions, layouts, I find it hard to immediately locate information on how to use the UI elements' class attributes, such as ui.xx.classes('xxx'). There's no clear guidance on what these classes should contain or how they should be used. This is far from user-friendly for beginners. Those who write the documentation seem to assume that users of NiceGUI are front-end experts. In reality, it's often newcomers to front-end development, like myself, who are more likely to use NiceGUI.

**Reply by zouzhe1** [reply]
![image](https://github.com/user-attachments/assets/032b9156-9b6e-4cd8-a8d7-77fe00ddb6dc)
Here is the documentation, it appears utterly disorganized, seemingly suitable only for someone who wishes to do nothing more than cycle through each option without a specific goal in mind.

**Reply by python-and-novella** [reply]
Would you have examples?
Or would like the style of textual?
https://textual.textualize.io/
Their document are very detailed and easy to learn for starter.

**Comment by falkoschindler** [comment]
With the current structure we tried to organize the long list of UI elements and functions into more manageable sections. It was the result of discussion #1807 and is generally well received by the community.

Have you seen the reference sections like https://nicegui.io/documentation/label#reference? There you'll find a complete list of class methods, properties etc.

And have you tried the website's search feature? It is a great way to find specific information in a large documentation like this.

**Reply by python-and-novella** [reply]
I always believe that using any tool that integrates multiple technologies without learning other technologies is abnormal. People can use AI to simplify the process of beginners, but we cannot treat lazy beginner habits as the norm.
NICEGUI is a great beginner tool, but if you want to fully customize, you still need to understand the front-end basics. HTML, CSS, Tailwindcss, Quasar, JavaScript are not only the foundation of NICEGUI, but also very useful knowledge that is worth spending time learning. Many people may think that NICEGUI is just a prototype tool, but they overlook the need for necessary front-end knowledge to beautify the basic prototype.

**Reply by falkoschindler** [reply]
The words "abnormal" and "lazy" sound a bit harsh...

I think it's very challenging to write (and maintain) a really good documentation. You need to address beginners as well as advanced programmers, some of which look for more textual descriptions, others prefer interactive demos. We do our very best and are open for constructive feedback. But there might not be the one documentation format that fits all.

**Reply by python-and-novella** [reply]
Sorry for that,I use machine interpretor.It makes me a little angry.
I have learnt lots of frameworks which design for python. Totally, Nicegui is not  the most  difficult or easy to learn.
I am not a front-end expert but know it a little. After I read document fully for 7 days, I'm learning CSS while I use NICEGUI.
To be honest, examples are better than API doc which is designed for professional programer.
So, I'd love to learn in discussions,although GITHUB is not easy to connect in China.
For beginner,document's demo code is easy to copy.If someone wants to learn more,discussions and search in document are better.
But basic knowledge is always the first.
 Nicegui is fun enough to start,we need to learn more before digging deeper.

**Comment by cargocultprogramming** [comment]
I just started to test nicegui and while I like the documentation, I agree kind of.

While the sections are good in order to find things you think will "probably" be in this section, it's extra effort to drill down to the "reference" section and that in itself seems incomplete. Right now for example I'm looking for the documentation for "app" and I can't find it, because search gives me too many results and the "logical" sections don't have it. It might be in a "reference" detail page, but how do I get there?

While experimenting with nicegui I also find myself constantly having _all_ section pages open in different tabs, so that I can find details on individual things faster. Otherwise I dive into a section, back, into another once, scroll down, go to the details page...

I'd love a format where you have the sections as they are now (nicely grouped and with short examples to see what things do), but instead of the individual reference pages something like the classic "api reference documentation" where I can look up all the details in a single place not just for the single, individual element.

**Reply by falkoschindler** [reply]
We also found that the NiceGUI documentation was missing a thorough overview of all modules and elements. Therefore we created the ["Map of NiceGUI"](https://nicegui.io/documentation#map-of-nicegui) which got released yesterday. It's supposed to show the big picture and links to individual documentation pages (if they exist). Maybe we'll find a way to integrate it more tightly into the documentation hierarchy in the future, maybe as some kind of side menu.

**Reply by cargocultprogramming** [reply]
That helps a lot! Good timing too! 😅

**Reply by Mte90** [reply]
that map is very useful!

If I had that documentation earlier It would saved a lot of time :-D
